metron-interface/metron-config/e2e/sensor-list/sensor-list.po.ts

```diff
@@ -0,0 +1,240 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions andx
+ * limitations under the License.
+ */
+
+import { browser, element, by, protractor } from 'protractor/globals';
+import { waitForElementPresence, waitForStalenessOf } from '../utils/e2e_util';
+var Promise = require('bluebird');
+
+export class SensorListPage {
+
+    clickOnActionsAndWait(parserNames: string[], clickOnClassName: string, waitOnClassName: string) {
+        let queueForClick = Promise.resolve();
+        parserNames.forEach((name) => {
+            queueForClick = queueForClick.then((result) => {
+                return this.getIconButton(name, clickOnClassName).click();
+            });
+        });
+
+        return queueForClick.then(() => {
+            let promiseArray = [];
+            parserNames.map(name => {
+                promiseArray.push(this.waitForElement(this.getIconButton(name, waitOnClassName)));
+            });
+
+            return protractor.promise.all(promiseArray).then(args => {
+                return args;
+            });
+        });
+    }
+
+    clickOnDropdownAndWait(parserNames: string[], dropDownLinkName: string, waitOnClassName: string) {
+        return protractor.promise.all([this.toggleSelectAll(), this.toggleDropdown()]).then(() => {
+
+            return element(by.css('span[data-action=\"'+ dropDownLinkName +'\"]')).click().then(() => {
+                let promiseArray = [];
+                parserNames.map(name => {
+                    promiseArray.push(this.waitForElement(this.getIconButton(name, waitOnClassName)));
```

Comment:
more efficient to avoid side-effects here, since map returns the result of a transformation, e.g. promiseArray = parserNames.map( (name) => x )